flask/src/main/groovy/net/corda/gradle/flask/FlaskPlugin.groovy

```diff
@@ -0,0 +1,67 @@
+package net.corda.gradle.flask
+
+import net.corda.flask.common.Flask
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.file.Directory
+import org.gradle.api.plugins.JavaApplication
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.plugins.JavaPluginConvention
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.JavaExec
+import org.gradle.api.tasks.SourceSet
+import org.gradle.jvm.tasks.Jar
+
+class FlaskPlugin implements Plugin<Project> {
+
+    @Override
+    void apply(Project project) {
```

Comment:
Our other plugins all contain a few lines like this:
```java
private static final String MINIMUM_GRADLE_VERSION = "x.y";

if (GradleVersion.current().compareTo(GradleVersion.version(MINIMUM_GRADLE_VERSION)) < 0) {
    throw new GradleException("The <BLAH!> plugin requires Gradle " + MINIMUM_GRADLE_VERSION + " or newer.");
}
```
Do you know what Flask's minimum Gradle version is, please? It must be _at least_ "6.0" because it uses [BasePluginConvention.getLibsDirectory()](https://docs.gradle.org/current/javadoc/org/gradle/api/plugins/BasePluginConvention.html#getLibsDirectory--).

Reply:
yeah, that should be correct... I could add an `if` to fall back to `getLibsDir` on Gradle 5.x if that is required (it looks a bit strict to ask for version 6.x only because of that)

Reply:
I don't think we need to do that unless we intend to migrate Corda 4.8+ to use `flask` too, which seems unlikely. And `quasar-utils` and `cordapp-cpk` will require 6.6 anyway.

Reply:
also, I think it would be anyway better to add that block in a static block in `FlaskJarTask` because you could happily use that task without actually applying the plugin (just adding it to the classpath)

Reply:
> ... you could happily use that task without actually applying the plugin (just adding it to the classpath)

That's true, but I have no idea precisely _when_ Gradle will decide to initialise `FlaskJarTask.class`. While you would be guaranteed to run the check if you _did_ use the task, I couldn't also say that you'd be guaranteed _not_ to run the check if you _didn't_ :thinking:.

Reply:
well [this line](https://github.com/woggioni/corda-gradle-plugins/blob/flask/flask/src/main/groovy/net/corda/gradle/flask/FlaskPlugin.groovy#L38) contains an instance of the `FlaskJarTask` class, which means the JVM has to load the class first (hence run the static block inside it)